### PR TITLE
convert zlinux lunid to the format in by-path folder

### PR DESCRIPTION
**Pull request by 27149chen**
convert zlinux lunid to the format in by-path folder like this:
ccw-0.0.a200-fc-0x5005076306005329-lun-0x4010401f00000000

### CODE REVIEW

**Comment by oriyarde** [comment]
> Is this for all CPU architectures? or just Z?

Also, is this for all storage types? of just ds8k?

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go [hunk 1]
```diff
@@ -144,7 +162,7 @@ func (r OsDeviceConnectivityHelperScsiGeneric) GetMpathDevice(volumeId string, l
 
 	for _, arrayIdentifier := range arrayIdentifiers {
 		dp := strings.Join([]string{targetPath, connectivityType, arrayIdentifier, "lun", lunIdStr}, "-")
-		logger.Infof("GetMpathDevice: Get the mpath devices related to connectivityType=%s initiator=%s and lunID=%s : {%v}", connectivityType, arrayIdentifier, lunIdStr, dp)
+		logger.Infof("GetMpathDevice: Get the mpath devices related to connectivityType=%s initiator=%s and lunID=%d : {%v}", connectivityType, arrayIdentifier, lunId, dp)
```
Comment (oriyarde) [inline node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: I think the log should still include somewhere the value after conversion (lunIdStr)

Comment (27149chen) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: yes, last params (dp) has it

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go [hunk 2]
```diff
@@ -118,8 +118,26 @@ func (r OsDeviceConnectivityHelperScsiGeneric) RescanDevices(lunId int, arrayIde
...
 }
 
+/*
+There are two style lun number, one's decimal value is <256 and the other
+is full as 16 hex digit. According to T10 SAM, when decimal value is more
+than 256 and it is converted to the full 16 hex digit, it should be
+swapped and converted into hex.
+
+see the udev function format_lun_number
+https://github.com/systemd/systemd/blob/c4ae2704b7e921a0b05486a7b201be6770a04ea7/src/udev/udev-builtin-path_id.c#L58
```
Comment (igorgonibm) [inline node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: i don't like such links. They may change in the future
f you want to leave it better write in more generic way "github systemd project file udev-builtin-path_id.c"

**Comment by 27149chen** [comment]
It is not releated to arch or storage types.

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go (hunk 1 above)
Comment (igorgonibm) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: I would like both before and after. (rawLunID=%s lunID=%s)

Comment (oriyarde) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: ok

**Comment by oriyarde** [comment]
@27149chen please add UT coverage for this bug to a ticket, same as:
https://github.com/IBM/ibm-block-csi-driver/pull/140#issuecomment-605635793

File: node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go (hunk 2 above)
Comment (27149chen) [reply node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go]: It is a blob object, it won't change.

**Comment by oriyarde** [comment]
@27149chen can we delete this branch?

**Pull request merged**